### PR TITLE
drivers: gpio: mec5: Microchip MEC5 HAL based GPIO driver

### PR DESCRIPTION
**Pull request by scottwcpg**
Add a GPIO driver for the Microchip MEC5 HAL based chips. Current devices are: MEC174x, MEC175x, and HAL version of MEC172x named MECH172x.

### CODE REVIEW

File: dts/bindings/gpio/microchip,mec5-gpio.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+include: [gpio-controller.yaml, base.yaml]
+
+properties:
+  reg:
+    required: true
+
+  port-id:
+    type: int
+    required: true
+    description: Zero based GPIO port number
```
Comment (henrikbrixandersen) [inline dts/bindings/gpio/microchip,mec5-gpio.yaml:17]: Port numbers seems to be a HAL'ism, not a property of the hardware IP itself. Please see if you can work-around having this set in the devicetree nodes.

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

File: dts/bindings/gpio/microchip,mec5-gpio.yaml (hunk 1 above)
Comment (scottwcpg) [reply dts/bindings/gpio/microchip,mec5-gpio.yaml]: Removed port-id and girq-id (unused) properties.

**Review by albertofloyd — APPROVED** [review]

**Comment by albertofloyd** [comment]
ping @henrikbrixandersen

**Review by jvasanth1 — APPROVED** [review]

**Comment by scottwcpg** [comment]
@nashif @MaureenHelm 
Request stale label be removed.
Do I need a third approval?

**Review by henrikbrixandersen — APPROVED** [review]

**Comment by henrikbrixandersen** [comment]
> @nashif @MaureenHelm 
> 
> Request stale label be removed.

I've removed it. 

> Do I need a third approval?

Sorry, it was pending my review.

**Comment by scottwcpg** [comment]
@henrikbrixandersen 
Thanks!

**Pull request merged**